### PR TITLE
cleanup: Remove deprecated values for KPR

### PR DESCRIPTION
**Pull request by sayboras**
## Description

This commit is to remove all deprecated values (strict, disabled, probe
and partial) for kubeProxyReplacement.

Relates: https://github.com/cilium/cilium/pull/26036, https://github.com/cilium/cilium/pull/26496

## Testing

Testing was done with the latest cilium-cli with commit hash while waiting for the next release.

Relates: https://github.com/cilium/cilium/pull/31286

### CODE REVIEW

File: .github/workflows/conformance-clustermesh.yaml
Comment (gandro) [inline .github/workflows/conformance-clustermesh.yaml]: Why are changes to `set-string` necessary? Previously we were accepting `kubeProxyReplacement` as either a bool or a string, thanks to `{{- $stringValueKPR := (toString .Values.kubeProxyReplacement) -}}`.

I think we should continue to accept plain booleans. I think only accepting `"true"` and `"false"` (string) instead of `true`/`false` (boolean) will lead to a bad user experience.

Comment (sayboras) [reply .github/workflows/conformance-clustermesh.yaml]: This is to make it work with current cilium-cli version, because the CLI always overrides it with strict. Using --set-string is a workaround in that case.

Probably need to wait for cilium-cli PR to land in first https://github.com/cilium/cilium-cli/pull/2365

**Comment by sayboras** [comment]
Marking this draft to wait for https://github.com/cilium/cilium-cli/pull/2365, then the change for --helm-set is no longer required.

**Review by michi-covalent — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (tklauser) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: nit:

```suggestion
  {{ fail "kubeProxyReplacement must be explicitly set to a valid value (true or false) to continue." }}
```

File: pkg/option/config.go
Comment (tklauser) [inline pkg/option/config.go]: Might make sense to adjust the godoc comments for these given that the referenced "strict" and "partial" values and their godoc no longer exist after this PR, e.g.

```go
// KubeProxyReplacementTrue specifies to enable all kube-proxy replacement
// features (might panic).
KubeProxyReplacementTrue = "true"

// KubeProxyReplacementFalse specifies to enable only selected kube-proxy
// replacement features (might panic).
KubeProxyReplacementFalse = "false"
```

**Review by tklauser — APPROVED** [review]
LGTM, two nits inline.

**Review by gandro — APPROVED** [review]
My CODEOWNERs look good to me, thanks!

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (sayboras) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml]: good catch, I will wait for other reviewers if there is any other changes, then will bundle them all.

File: pkg/option/config.go
Comment (sayboras) [reply pkg/option/config.go]: Thanks, will bundle all review comments together later.

File: Documentation/operations/upgrade.rst
Comment (qmonnet) [inline Documentation/operations/upgrade.rst]: I would maybe summarise the differences between the option sets (if I remember: `true` == `strict`, `false` is like `disabled` but you can turn on individual features yourself) or at least link to the relevant docs (changed in https://github.com/cilium/cilium/commit/1a4e7d442543b5b09756c5ef2d0a5e68393f1020).

(Same for Helm.)

File: daemon/cmd/kube_proxy_replacement_test.go
Comment (qmonnet) [inline daemon/cmd/kube_proxy_replacement_test.go:135]: Why not testing this with `false` instead?

**Review by nathanjsweet — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Review by gentoo-root — APPROVED** [review]

**Review by brb — APPROVED** [review]
Thanks!

**Review by squeed — APPROVED** [review]

File: Documentation/operations/upgrade.rst
Comment (danehans) [reply Documentation/operations/upgrade.rst]: +1

File: daemon/cmd/kube_proxy_replacement_test.go
Comment (danehans) [reply daemon/cmd/kube_proxy_replacement_test.go:135]: +1

**Review by danehans — APPROVED** [review]
Approving with the understanding that @qmonnet @tklauser review comments will be resolved before merging.

File: Documentation/operations/upgrade.rst
Comment (sayboras) [reply Documentation/operations/upgrade.rst]: Good point, I have added the reference pointing to KPR page :+1:

File: daemon/cmd/kube_proxy_replacement_test.go
Comment (sayboras) [reply daemon/cmd/kube_proxy_replacement_test.go:135]: good point. I have adjust this test and move it close to other KPR=false test.

**Review by qmonnet — APPROVED** [review]
Thanks Tam!

File: Documentation/operations/upgrade.rst
Comment (aanm) [inline Documentation/operations/upgrade.rst]: Small nit: Please use ``true`` **or** ``false`` instead.

**Review by aanm — APPROVED** [review]
I was worried about the [set-strings](https://github.com/cilium/cilium/pull/31286#discussion_r1519714720) at first. I'm glad it got fixed in the meanwhile

File: Documentation/operations/upgrade.rst
Comment (sayboras) [reply Documentation/operations/upgrade.rst]: sorry, I don't get the above comment :thinking:. Do we have the same line here?

> Please use `true` or `false` instead

Comment (sayboras) [reply Documentation/operations/upgrade.rst]: ah ignore me, I cannot read logically, it's `or` instead of `and`

File: Documentation/operations/upgrade.rst
Comment (youngnick) [inline Documentation/operations/upgrade.rst]:
```suggestion
  removed. Please use ``true`` or ``false`` instead. Please refer to :ref:`kube-proxy replacement <kubeproxy-free>`
```

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (youngnick) [inline install/kubernetes/cilium/values.yaml.tmpl]:
```suggestion
# Valid options are "true" or "false".
```

File: Documentation/operations/upgrade.rst
Comment (youngnick) [reply Documentation/operations/upgrade.rst]: I think you missed this one, sorry.

Comment (sayboras) [reply Documentation/operations/upgrade.rst]: let me check and send quick PR, good thing is that docs change will not require full CI run.

Comment (sayboras) [reply Documentation/operations/upgrade.rst]: thanks and done

**Comment by sayboras** [comment]
The last force push is only doc change, I don't think full ci is required.

https://github.com/cilium/cilium/compare/f3b9be46f458a2c42794d3d12bfef2a9f829b5a9..e17e1239a2e160bf166d824867f3c22f4672b35c

**Review by youngnick — APPROVED** [review]

**Pull request merged**
